.github/workflows/dependency-review.yml

```diff
@@ -0,0 +1,20 @@
+# Dependency Review Action
+#
+# This Action will scan dependency manifest files that change as part of a Pull Request, surfacing known-vulnerable versions of the packages declared or updated in the PR. Once installed, if the workflow run is marked as required, PRs introducing known-vulnerable packages will be blocked from merging.
+#
+# Source repository: https://github.com/actions/dependency-review-action
+# Public documentation: https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement
+name: 'Dependency Review'
```

Comment:
How does this actually work, what does it do? From the [README](https://github.com/actions/dependency-review-action#dependency-review-action) it is not clear what exactly is being reviewed.

Reply:
Let's try it and find out. It isn't like we cannot remove it.

Reply:
It works on a PR and checks if the PR adds or updates any dependencies. If so, it checks if any of those dependencies have a known vulnerability as per the GitHub Advisory Database. 

I created a POC to demo this. Here is a PR that adds dependencies using package.json (same concept applies to other package registries as well). https://github.com/varunsh-coder/test/pull/5

One of the packages has a known vulnerability, so the workflow run fails. 
https://github.com/varunsh-coder/test/actions/runs/4465005220/jobs/7841664492#step:5:9